app/models/user.rb

```diff
@@ -0,0 +1,80 @@
+class User < ActiveRecord::Base
+  has_many :products
+  has_many :order_items
+
+  has_secure_password
+
+  validates :first_name, presence: true
+  validates :first_name, length: { :minumum => 1, :maximum => 30 }
```

Comment:
Not sure about these 30 character first / last name maximums. It seems unnecessarily restrictive and could restrict folks with really long names from signing up.